src/libvirt/LibvirtObject.cs

```diff
@@ -51,13 +51,13 @@ protected int GetInt32(Func<int> func)
             return result;
         }
 
-        protected void ThrowExceptionOnError(int result) =>  ThrowExceptionOn(() => result == -1);
+        protected internal static void ThrowExceptionOnError(int result) => ThrowExceptionOn(() => result == -1);
 
-        protected void ThrowExceptionOnError(IntPtr result) => ThrowExceptionOn(() => result == IntPtr.Zero);
+        protected static void ThrowExceptionOnError(IntPtr result) => ThrowExceptionOn(() => result == IntPtr.Zero);
 
-        protected void ThrowExceptionOnError(string result) => ThrowExceptionOn(() => result == null);
+        protected static void ThrowExceptionOnError(string result) => ThrowExceptionOn(() => result == null);
 
-        private void ThrowExceptionOn(Func<bool> predicate)
+        private static void ThrowExceptionOn(Func<bool> predicate)
```

Comment:
We should find a better design. I would not just make these functions static. If they are helpers not only for LibvirtObject but also for libvirt.cs, I would move them in some static helper class (e.g. LIbvirtHelper?)